kotlin-native/backend.native/tests/objcexport/coroutines.swift

```diff
@@ -48,6 +48,39 @@ private func testUnitCallSimple() throws {
     try assertNil(error)
 }
 
+private func testUnitCallNonMainDispatcher() throws {
+#if LEGACY_SUSPEND_UNIT_FUNCTION_EXPORT
+    var result: KotlinUnit? = nil
+#endif
+    var error: Error? = nil
+    var completionCalled = 0
+
+    let group = DispatchGroup()
+
+#if ALLOW_SUSPEND_ANY_THREAD
+#if LEGACY_SUSPEND_UNIT_FUNCTION_EXPORT
+    DispatchQueue.global().async(group: group) {
+        CoroutinesKt.unitSuspendFun { _result, _error in
+            completionCalled += 1
+            result = _result
+            error = _error
+        }
+    }
+#else
+    DispatchQueue.global().async(group: group) {
+        CoroutinesKt.unitSuspendFun { _error in
+            completionCalled += 1
+            error = _error
+        }
+    }
+#endif
+#endif
+
+    group.wait()
+    try assertNil(error)
```

Comment:
```suggestion
    try assertEquals(actual: completionCalled, expected: 1)
    try assertNil(error)
```

Reply:
done, had to also update the test to work in the else case of the ifdef where the flag isn't passed.